File: dataset_reduzido/avatar/Python/Code/atcoder_AGC024_B.py
def main ( ) :
    N = int ( input ( ) )
    P = [ int ( input ( ) ) for _ in range ( N ) ]
    editorial ( N , P )
def editorial ( N , P ) :
    if N == 1 :
        print ( 0 )
        return
    a = [ 0 ] * ( N + 1 )
    for i , p in enumerate ( P ) :
        a [ p ] = i
    tmp = 1
    max_len = 1
    for i in range ( 1 , N ) :
        if a [ i ] < a [ i + 1 ] :
            tmp += 1
            max_len = max ( max_len , tmp )
        else :
            tmp = 1
    ans = N - max_len
    print ( ans )
def WA ( N , P ) :
    tmp = 0
    ans = 0
    for i , p in enumerate ( P ) :
        if i == 0 or P [ i - 1 ] + 1 == p :
            tmp += 1
        else :
            ans = max ( ans , tmp )
            tmp = 1
    print ( N - ans )
if __name__ == '__main__' :
    main ( )

File: output/Llama-2-100/avatar/Python/Java/atcoder_AGC024_B.java
public class atcoder_AGC024_B {
    public static void main(String[] args) {
        int N = Integer.parseInt(sc.nextLine());
        int[] P = new int[N];
        for (int i = 0; i < N; i++) {
            P[i] = Integer.parseInt(sc.nextLine());
        }
        editorial(N, P);
    }

    public static void editorial(int N, int[] P) {
        if (N == 1) {
            System.out.println(0);
            return;
        }
        int[] a = new int[N + 1];
        for (int i = 0; i < N; i++) {
            a[P[i]] = i;
        }
        int maxLen = 1;
        for (int i = 1; i < N; i++) {
            if (a[i] < a[i + 1]) {
                maxLen = Math.max(maxLen, 1 + a[i]);
            } else {
                maxLen = 1;
            }
        }
        int ans = N - maxLen;
        System.out.println(ans);
    }

    public static void WA(int N, int[] P) {
        int ans = 0;
        for (int i = 0; i < N; i++) {
            if (i == 0 || P[i - 1] + 1 == P[i]) {
                ans++;
            } else {
                ans = Math.max(ans, 1 + ans);
            }
        }
        System.out.println(N - ans);
    }
}


